src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -359,17 +360,32 @@ void remove_virtual_functionst::get_functions(
   }
 
   // iterate over all children, transitively
-  std::set<irep_idt> visited;
   get_child_functions_rec(
     class_id,
     root_function.symbol_expr,
     component_name,
     functions,
-    visited,
     resolve_function_call);
 
   if(root_function.symbol_expr!=symbol_exprt())
     functions.push_back(root_function);
+
+  // remove all classes from dispatch table where concrete call has been found
```

Comment:
I believe this currently filters nothing, as the set contains those entries that resolved to a function other than the root, and the filter condition requires membership of that set *and* having resolved to the root function.